addons/account/models/account_move.py

```diff
@@ -3569,8 +3569,29 @@ def button_set_checked(self):
             move.to_check = False
 
     def button_draft(self):
+        exchange_move_ids = set()
+        if self:
+            self.env['account.full.reconcile'].flush_model(['exchange_move_id'])
+            self.env['account.partial.reconcile'].flush_model(['exchange_move_id'])
+            self._cr.execute(
+                """
+                    SELECT exchange_move_id
+                    FROM account_full_reconcile
+                    WHERE exchange_move_id IN %s
+
+                    UNION ALL
+
+                    SELECT exchange_move_id
+                    FROM account_partial_reconcile
+                    WHERE exchange_move_id IN %s
+                """,
+                [tuple(self.ids), tuple(self.ids)],
+            )
+            for row in self._cr.fetchall():
+                exchange_move_ids.add(row[0])
```

Comment:
Can't you just set a `SELECT DISTINCT exchange_move_id FROM (subquery)` so we don't need to create a set for this.